sql/core/src/main/scala/org/apache/spark/sql/internal/SharedState.scala

```diff
@@ -82,6 +82,19 @@ private[sql] class SharedState(val sparkContext: SparkContext) extends Logging {
    */
   val cacheManager: CacheManager = new CacheManager
 
+  /**
+   * A status store to query SQL status/metrics of this Spark application, based on SQL-specific
+   * [[org.apache.spark.scheduler.SparkListenerEvent]]s.
+   */
+  val statusStore: SQLAppStatusStore = {
```

Comment:
I just realized this is important. Previously we can use `SharedState.listener` to query the SQL status. `SharedState` is actually a semi-public interface, as it's marked as `Unstable` in `SparkSession.sharedState`.

Sometimes for debugging I just type `spark.sharedState.listener.xxx` in Spark Shell and check some status, but it's impossible now after #19681 

There might be some other people like me that love this ability, we should not just remove it for future UI discoverbility(I think it's just SQL and streaming, really not a big gain)

cc @hvanhovell @viirya @gatorsmile

Reply:
+1, this is behavior change. It was good to have `SharedState.listener` and check metrics from it.

Reply:
+1

Reply:
Makes sense to me.

Reply:
Even if you add the property back, it won't be the same listener. There's no way to keep the old API without keeping all of the old code.

Reply:
at least it's developer-facing, as a developer I don't care about the naming changing, or API changing, but I just want the same functionality.

Reply:
Sure, it's fine if you want to expose it. But I'm pointing out that it's pretty weird to expose a class in a ".internal" package through the API. Those are not documented nor go through mima checks, so there's absolutely zero guarantees about them.

Reply:
That's how it works, those things are not critical to the end users and it's ok to break them if no one objects.

Reply:
Also this is consistent to `SparkContext.statusStore`, we need central places to query the core/sql/streaming status.

Reply:
I'd use `@Private` instead of `@Unstable` for the SparkSession method, that's all I'm saying. It more clearly maps to what you and the code are saying.